### PR TITLE
[Bootstrap] Fix spaces in Xcode issue

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
sdk and compiler path needed escaping.

- https://bugs.swift.org/browse/SR-2650
- <rdar://problem/28318153> SR-2650 Can't build swiftpm with using Xcode
that has a whitespace in its name

### CODE REVIEW

File: Utilities/bootstrap [hunk 1]
```diff
@@ -476,10 +476,10 @@ def create_bootstrap_files(sandbox_path, args):
         elif target.is_swift and not target.is_library:
             link_output_path = os.path.join(bin_dir, target.name)
 
-            link_command = [args.swiftc_path,
+            link_command = [pipes.quote(args.swiftc_path),
                             '-o', pipes.quote(link_output_path)]
             if args.sysroot:
-                link_command.extend(["-sdk", args.sysroot])
+                link_command.extend(["-sdk", pipes.quote(args.sysroot)])
```
Comment (ddunbar) [inline Utilities/bootstrap]: This doesn't feel right, something else must be wrong here... we should be passing each argument in this array literally.

I think the actual probably might be in how we pass this command to swift-build-tool, maybe we are using the old arguments form?

Comment (aciidb0mb3r) [inline Utilities/bootstrap]: Yes its using the old arguments form but using array literal did not work because of some commands were separated by a `;`

Comment (aciidb0mb3r) [inline Utilities/bootstrap]:
```
    args: rm -f /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/lib/PackageDescription.a ; ar cr /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/lib/PackageDescription.a /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/PackageDescription.build/JSON.o /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/PackageDescription.build/Package.o /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/PackageDescription.build/Product.o /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/PackageDescription.build/Target.o /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/PackageDescription.build/Version+StringLiteralConvertible.o /Users/ankit/workspace/opensource/swift.org/swiftpm/.build/.bootstrap/PackageDescription.build/Version.o

```

Comment (ddunbar) [inline Utilities/bootstrap]: Good news: llbuild now has an `archive` tool which embeds this logic, so we can just switch to that.

**Comment by aciidb0mb3r** [comment]
@ddunbar done

**Comment by abertelrud** [comment]
LGTM

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test linux

**Comment by ddunbar** [comment]
@swift-ci please test linux

**Comment by ddunbar** [comment]
LGTM

**Comment by aciidb0mb3r** [comment]
@swift-ci please test linux

**Pull request merged**